fccodegen/_generator.py

```diff
@@ -158,39 +273,86 @@ def derivative_name(deriv):
         return 'd_dt_' + unames[var]
 
     # Create expression printer
-    p = cg.WebLabPrinter(symbol_name, derivative_name)
-
-    #TODO: Update cellmlmanip to make this code look more like:
-    #
-    # state_vars = model.get_state_variables()
-    # free_var = model.get_free_variable()
-    # derivatives = [sp.Derivative(state, free_var) for state in state_vars]
-    # eqs = model.get_equations_for(derivatives)
-    #
-
-    # Create a list of state variable indices, variables, and derivative vars
-    states = []
-    for i, state in enumerate(get_state_list(graph)):
-        states.append({
+    printer = cg.WebLabPrinter(symbol_name, derivative_name)
+
+    # Create free variable name
+    free_name = symbol_name(get_free_variable_symbol(graph))
+
+    # Create state information dicts
+    state_info = []
+    for i, state in enumerate(get_state_symbols(graph)):
+        #TODO: Update cellmlmanip so this graph.nodes goes away
+        initial_value = float(graph.nodes[state]['initial_value'])
+        state_info.append({
             'index': i,
             'var_name': symbol_name(state),
             'deriv_name': derivative_name(state),
+            'initial_value': initial_value,
+        })
+
+    # Create parameter information dicts
+    parameter_info = []
+    for i, parameter in enumerate(parameters):
+        symbol = get_symbol_by_cmeta_id(graph, parameter)
```

Comment:
This will need to be an RDF lookup eventually, but I think you're aware of that :)

Reply:
Have made a note of it in the code too now!